Sources/third_party/Experimental/Complex.swift

```diff
@@ -0,0 +1,362 @@
+// Copyright 2017-2019 Xiaodi Wu and The TensorFlow Authors. All Rights Reserved.
+//
+// Licensed under the MIT License (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://opensource.org/licenses/MIT
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//  Note
+//  ====
+//
+//  For maximum consistency with corresponding functions in C/C++, checks for
+//  special values in `naturalExponential()`, `squareRoot()`, trigonometric
+//  functions, and hyperbolic functions are adapted from libc++.
+//
+//  Code in libc++ is dual-licensed under the MIT and UIUC/NCSA licenses.
+//  Copyright © 2009-2017 contributors to the LLVM/libc++ project.
+/// A type to represent a complex value in Cartesian form.
+///
+/// - Note: `Complex64` is a type alias for `Complex<Float>` and `Complex128` is
+///   a type alias for `Complex<Double>`.
+///
+/// Create new instances of `Complex<T>` using integer or floating-point
+/// literals and the imaginary unit `Complex<T>.i`. For example:
+///
+/// ```swift
+/// let x = 2 + 4 * .i // `x` is of type `Complex<Double>`
```

Comment:
I don't think this will type infer correctly given that there is no contextual complex type

Reply:
Ahh yes you're right I would need to add more context like `let x: Complex<Double> = 2 + 3 + .i`

I added this documentation back when @rxwei pointed out that I should keep the original header by Xiaodi. However, I think I skimmed over to quickly when I read the following:

> For files in the third_party dir, leave the existing headers in place. If you’ve modified a file, append “Google LLC” to the copyright line.

in the [google third-party info](https://opensource.google.com/docs/releasing/preparing/#third-party-components) (and also I didn't closely read the original header and missed this typo you caught). 

Since the third-party use is only about the license part, and not the entire original documentation provided by the author at the top of the file, I could remove this all, but I think it is best to keep it all and fix this typo.